src/background.ts

```diff
@@ -20,11 +20,27 @@ let currentHostname: string;
 
         listenToTabStatusChange();
         listenToRuntimeMessages();
+        listenToIdleTimer();
     } catch (error) {
         console.error(error);
     }
 })();
 
+function listenToIdleTimer(){
+    chrome.idle.setDetectionInterval(15);
+
+    chrome.idle.onStateChanged.addListener((newState) => {
+        if (newState != 'active' && intervalId){
+            clearInterval(intervalId);
+            intervalId = null;
```

Comment:
Why set it to null??

Reply:
clearInterval() with invalid ID will clear all intervals. To prevent this, we reset intervalId.